docs/specifications/tests/Connectivity-TP/connectivity03.md

```diff
@@ -1,67 +1,214 @@
 
-The objective in this test is to check that all IP addresses of the domain's
-authoritative name servers does not belong to the same AS.
+This test is done separately on IPv4 and IPv6, and both must match the criterion.
 
-[missing references to AS numbers, RFC 1930?]
+[RFC 2182], section 3.1 clearly specifies that distinct authoritative name 
+servers for a child domain should be placed in different topological and 
+geographical locations. The objective is to minimise the likelihood of a single 
+failure disabling all of them. 
 
-### Inputs
 
-The domain name to be tested.
+## Inputs
 
-### Ordered description of steps to be taken to execute the test case
+* The domain name to be tested ("child zone");
+* The ASN database to be used ("RIPE database" or "Cymru database").
+* The base name (Cymru) or Whois server (RIPE) to be used.
 
-1. Obtain the IP addresses of the name servers from [Method4](../Methods.md).
-2. Obtain the list of ASN lookup domains from the configuration of Zonemaster.
-3. For obtaining the AS numbers for IPv6 addresses; first reverse its eight
-4-character hexadecimal numbers and place dots between each. It is important 
-all omitted zeroes are included.  
-3.1. At the end of the reversed IPv6 address, concatenate the string (from
-step 2)  
-3.2 Send a "TXT" query using the string (obtained from Step 3.1)  
-3.3 If there is an ANSWER, then go to step 3.4, else go to the next string
-in the list (from step 2)  
-3.4 The AS number for the IPv6 address is found in the ANSWER for the query 
-4. To obtain the AS numbers for IPv4 addresses; first reverse the four numerical
-octets and place dots between each.  
-4.1. At the end of the reversed IPv4 address concatenate the string (from step 2)  
-4.2 Send a "TXT" query using the string (obtained from Step 4.1)  
-4.3 If there is an ANSWER, go to step 4.4, else go to the next string
-in the list (from step 2)  
-4.4 The AS number for the IPv4 address is found in the ANSWER for the query 
-5. If all the retrieved AS (obtained from step 3.3 and 4.3) are same, then the test
-   fails.
 
-### Outcome(s)
+## Ordered description of steps to be taken to execute the test case
 
-If there is a AS which is different from the other retrieved AS, then the
-test succeeds.
+1. Obtain the total set of IP addresses of the name servers for the 
+   *child zone* using [Method4] and [Method5].
 
-### Special procedural requirements
+2. For each IP address in the set from step 1 above, determine the ASN set
+   announcing the IP address using either the *[Cymru database]* or the 
+   *[RIPE database]* as described in separate sections below. 
 
-None
+3. For each IP protokoll (IPv4, IPv6) do:
+   1. If all addresses are announced from one and the same ASN emit
+      *[IPV4_ONLY_ONE_ASN]* or *[IPV6_ONLY_ONE_ASN]*.
+   2. If all addresses are announced from the same ASN set but from more than 
+      one ASN emit *[IPV4_SAME_ASN]* or *[IPV6_SAME_ASN]*.
```

Comment:
What do you think of this wording? "If all addresses are announced from the same ASN set of multiple ASNs."